vault/resource_pki_secret_backend_role.go

```diff
@@ -22,56 +25,103 @@ var (
 	pkiSecretBackendRoleNameFromPathRegex    = regexp.MustCompile("^.+/roles/(.+)$")
 )
 
+// Any new fields should probably not be added to these maps. Instead handle
+// them separately within a provider.IsAPISupported guard
```

Comment:
Wondering what people think of this suggestion?

In theory, we should be able to add new fields to these maps as long as we used `the provider.IsAPISupported` guard inside the for loops. I was thinking that this suggestion I am making above would be simpler though. 🤔

Reply:
This is just because there's lists of the fields that get looped through because of the large amount of them? The note makes sense to me, as to not add new fields to the existing lists since they won't be supported by the same API version as the others being looped through.

You called them "maps" in the comments, did you mean just "slices" or "lists"?

Reply:
woops... yep slices!

Reply:
I like this note! Should help incoming developers understand the code style and patterns 👍🏼